code/__defines/gamemode.dm

```diff
@@ -25,6 +25,8 @@
 #define BE_MUTINEER   0x2000
 #define BE_PAI        0x4000
 #define BE_LOYALIST   0x8000
+#define BE_HACKER     0x1F
```

Comment:
From what I know.. This SHOULD work.

Reply:
This is insane. Why are the roles even assigned as HEX 0x SHIT ANYWAY

Reply:
Bitfields are usually more memory efficient than storing things in a list, needing to iterate over every item in the list to see if something is present.

But, that's why I say to look at what /tg/ or Para or someone does, because they probablty have more than 16 pref things too and probably came up with another solution.

Reply:
I just got a computer lesson, and I [i]kind of[/i] understood it. Basically, doing things in hexadecimal is a lot easier for the system. But.. If this is the case, then is there any way add these two extra roles..?

Reply:
If you swapped the system to use integers, then you could add up to 2**16 types, but you'd need to refactor everything that uses this pref to check integers instead of bitfields, which is beyond the scope of this PR and would need to be made in a separate pr that has to be merged before this one.
If you continue to use bitfields, you could remove obsolete prefs, or you can add a second pref variable that's selected instead. I know rev/mutiny has been reworked so you can remove one or maybe two of those. You might be able to remove the malf pref, but I'd get confirmation on that from other devs.

Reply:
How do you differentiate whether something is checking for bit fields or integers? What's an example of something that checks the flag for bit fields?

Reply:
`if(varname & FLAG)` is usually something using bitfields, it's pretty common to #define the specific flags.
`if(varname == CONST)` is an integer check.
If you're looking for specific examples of bitfield application, I did a lot of bitfield stuff in #5161 , and will be fiddling with it some more this weekend.